[ArtillerySun] iP

ArtillerySun opened the pull request:
### ArsenBot

>“Your mind is for having ideas, not holding them.” – David Allen 

ArsenBot frees your mind of having to remember things you need to do. It's,

+ text-based
+ easy to learn
+ ~~FAST~~ SUPER FAST to use

All you need to do is,

1. download it from [here](https://github.com/ArtillerySun/ip).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Commit:
Add Gradle support

Eclipse-Dominator committed:
Bump gradle and lib version

aureliony committed:
build.gradle: Prevent generating a second JAR file

In build.gradle, the dependencies on distZip and/or distTar causes
the shadowJar task to generate a second JAR file for which the
mainClass.set("seedu.duke.Duke") does not take effect.
Hence, this additional JAR file cannot be run.
For this product, there is no need to generate a second JAR file
to begin with.

Let's remove this dependency from the build.gradle to prevent the
shadowJar task from generating the extra JAR file.

ArtillerySun committed:
level 0

ArtillerySun committed:
level 1

ArtillerySun committed:
level 2

ArtillerySun committed:
level 3

ArtillerySun committed:
level 4

ArtillerySun committed:
level 5

ArtillerySun committed:
level 6

ArtillerySun committed:
A-TextUiTesting

ArtillerySun committed:
modify for level 7

ArtillerySun committed:
Revert "modify for level 7"

This reverts commit 52bc3636220cd17d47eaac028a40f91bd46fafbb.

ArtillerySun committed:
no message

ArtillerySun committed:
level 7 finalise

ArtillerySun committed:
no message

ArtillerySun committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	data/history.txt

ArtillerySun committed:
modify the Deadline and Event class to fit the requirement of parsing string into time

ArtillerySun committed:
modify the task-related class to support LocalDateTime

ArtillerySun committed:
finish level-8

ArtillerySun committed:
clean code of main class

ArtillerySun committed:
modify the main class to the final level-8

ArtillerySun committed:
Merge branch 'branch-Level-8'

# Conflicts:
#	data/history.txt
#	src/main/java/ArsenBot.java

ArtillerySun committed:
modify the code into more OOP way

ArtillerySun committed:
fix the bug in class event

ArtillerySun committed:
Merge branch 'A-MoreOOP'

ArtillerySun committed:
rearrange the project into packages

ArtillerySun committed:
Merge branch 'A-Packages'

ArtillerySun committed:
Merge remote-tracking branch 'origin/add-gradle-support'

FooChao commented on src/main/java/arsenbot/command/AddCommand.java:
```diff
@@ -0,0 +1,47 @@
+package arsenbot.command;
+
+import arsenbot.storage.Storage;
+import arsenbot.task.*;
```
Our course don't allow .*

FooChao commented on src/main/java/arsenbot/storage/Storage.java:
```diff
@@ -0,0 +1,60 @@
+import arsenbot.task.Task;
+import arsenbot.task.Todo;
+
+import java.io.*;
```
Our course don't allow .*

FooChao reviewed (commented):
Overall ,in terms of coding standard, it is following the java coding standard well. 
The only nitpicks I have are the ".*" as mentioned in 2 different parts of your code.
Do add config/checkstyle/checkstyle.xml and config/checkstyle/suppressions.xml so that u can ensure that it is following the standard of CS2103T.
More information can be found here : 
https://se-education.org/guides/tutorials/checkstyle.html

ArtillerySun committed:
set up gradle

ArtillerySun committed:
Merge branch 'A-Gradle'

thenosewizard commented on src/main/java/arsenbot/storage/Storage.java:
```diff
@@ -0,0 +1,60 @@
+import java.util.List;
+
+public class Storage {
+
+    private final String filePath;
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
```
Do remember to write a JavaDoc description for your classes!

thenosewizard commented on src/main/java/arsenbot/task/Task.java:
```diff
@@ -0,0 +1,30 @@
+
+public abstract class Task {
+    protected String description;
+    protected boolean isDone;
```
Good that your boolean variable sound like booleans 👍

thenosewizard commented on src/main/java/arsenbot/command/UnmarkCommand.java:
```diff
@@ -0,0 +1,32 @@
+package arsenbot.command;
+
+import arsenbot.storage.Storage;
+import arsenbot.task.Task;
+import arsenbot.task.TaskList;
+import arsenbot.task.TaskManagerException;
+import arsenbot.ui.Ui;
+
+import java.io.IOException;
```
Good that you did not use wildcards in your import statements 👍

thenosewizard commented on src/main/java/arsenbot/task/Event.java:
```diff
@@ -0,0 +1,28 @@
+    }
+
+    @Override
+    public String toFileFormat() {
```
Naming conventions followed 👍

thenosewizard reviewed (commented):
LGTM 👍 . Good that you have followed all of the naming conventions provided by the course.

ArtillerySun committed:
add testing of 2 classes, TaskList and Storage.

ArtillerySun committed:
Merge branch 'A-JUnit'

ArtillerySun committed:
modify the build.gradle file to fit jar generation

ArtillerySun committed:
Merge branch 'A-Jar'

ArtillerySun committed:
comments write: add header comments on most classes

ArtillerySun committed:
Storage class: modify the import files to comply the coding standard

ArtillerySun committed:
Add FindCommand class
Parser class: add branch to handle find command
Ui class: add method to show found takes for command find
TaskList class: add method to find all tasks with keyword

ArtillerySun committed:
Merge branch 'branch-A-JavaDoc'

ArtillerySun committed:
Merge branch 'branch-A-CodingStandard'

ArtillerySun committed:
Merge branch 'branch-Level-9'

ArtillerySun committed:
Add a GUI to ArsenBot
Use the JavaFX technology to implement the GUI

ArtillerySun committed:
Merge branch 'branch-Level-10'

ArtillerySun committed:
feat: Add assertions to Ui class

- Added assertions in the `showError` method to ensure that error messages are not null before displaying.
- Added assertions in the `readCommand` method to ensure that the user input is not null or empty.
- Added assertions in `showTaskAdded` to check that tasks are not null and the task list size is valid before performing operations.
- Added assertions in the `showTaskList` method to ensure that the task list is not null and contains at least one task before displaying the tasks.
- These assertions document assumptions and help catch potential issues during runtime.

feat: Add assertions to Parser class

- Added assertions in the `parse` method to ensure that user input is not null or empty before processing.

feat: Add assertions to Storage class

- Added an assertion in the `load` method to ensure that the file path is not null before attempting to load tasks.
- Added assertions to verify that each task parsed from the file has a valid type and structure before adding it to the task list.
- In the `save` method, added assertions to check that the list of tasks is not null before saving to the file.
- These assertions help ensure the integrity of the task data when loading from and saving to storage, reducing the risk of file-related errors.

feat: Add assertions to DeleteCommand class

- Added an assertion in the `execute` method to verify that the task index is within the valid range before attempting to delete the task.

ArtillerySun committed:
Refactor ArsenBot class for improved error handling and clarity

- Simplified error handling in the constructor by using a try-catch block.
- Added detailed comments to explain the purpose and functionality of methods.
- Refactored `run()` method to enhance readability and maintainability.
- Improved `getResponse()` method to handle exceptions and return error messages clearly.

Refactor Parser class for better command handling

- Improved error message for unrecognized commands.
- Updated documentation to clarify the purpose of the `parse()` method.

Refactor AddCommand class to improve task creation logic

- Extracted task creation logic into separate methods for ToDo, Deadline, and Event.
- Introduced helper methods `extractDescription()` and `extractParts()` to handle input extraction and validation.
- Consolidated error handling and validation within dedicated methods.
- Enhanced documentation with detailed comments explaining the purpose of each method.

ArtillerySun committed:
Refactor Storage Class to Utilize Java Streams for Task Loading

ArtillerySun committed:
Merge pull request #2 from ArtillerySun/branch-A-CodeQuality

Refactor ArsenBot class for improved error handling and clarity

ArtillerySun committed:
Merge branch 'master' into branch-A-Assertions

ArtillerySun committed:
Merge pull request #3 from ArtillerySun/branch-A-Assertions

feat: Add assertions to Ui class

ArtillerySun committed:
Merge branch 'master' into branch-A-Streams

ArtillerySun committed:
Merge pull request #4 from ArtillerySun/branch-A-Streams

Refactor Storage Class to Utilize Java Streams for Task Loading

ArtillerySun committed:
Add help command for user guidance

ArtillerySun committed:
Merge branch 'branch-C-Help'

ArtillerySun committed:
Modify phrases used by the chatbot and the GUI

ArtillerySun committed:
Modify phrases used by chatbot and the GUI

ArtillerySun committed:
Merge branch 'branch-A-Personality'

ArtillerySun committed:
Modify the MainWindow class, Main class, and DialogBox class to make the naming more proper

ArtillerySun committed:
Update README.md to include JAR file path and improve user instructions

ArtillerySun committed:
Merge branch 'branch-A-UserGuide'

ArtillerySun committed:
Update README.md

ArtillerySun committed:
Update docs/README.md

ArtillerySun committed:
Update README.md

ArtillerySun committed:
Update README.md

ArtillerySun committed:
Main class: improve code quality

ArtillerySun committed:
MainWindow class: improve code quality

ArtillerySun committed:
Ui class: improve code quality

The Ui class has been refactored to improve consistency in return
types and enhance overall readability. Previously, several methods
mixed void return types with string outputs.
